prompt/git-k8s.nu

```diff
@@ -0,0 +1,438 @@
+def bright_cyan [] {
+  each { |it| $"(ansi -e '96m')($it)(ansi reset)" }
+}
+
+def bright_green [] {
+  each { |it| $"(ansi -e '92m')($it)(ansi reset)" }
+}
+
+def bright_red [] {
+  each { |it| $"(ansi -e '91m')($it)(ansi reset)" }
+}
+
+def bright_yellow [] {
+  each { |it| $"(ansi -e '93m')($it)(ansi reset)" }
+}
+
+def green [] {
+  each { |it| $"(ansi green)($it)(ansi reset)" }
+}
+
+def red [] {
+  each { |it| $"(ansi red)($it)(ansi reset)" }
+}
+
+# Get the current directory with home abbreviated
+def related [sub dir] {
+    if $sub == $dir {
+        return { related: '=', path: '' }
+    }
+    let suffix = (do --ignore-errors { $sub | path relative-to $dir })
+    if ($suffix | is-empty) {
+        return { related: '>', path: '' }
+    } else {
+        return { related: '<', path: $suffix}
+    }
+}
+
+export def "pwd_abbr" [] {
+  let pwd = ($env.PWD)
+
+  let to_home = (related $pwd $nu.home-path)
+
+  let cwd = if $to_home.related == '=' {
+      "~"
+  } else if $to_home.related == '>' {
+      $pwd
+  } else {
+      $'~(char separator)($to_home.path)'
+  }
+
+  mut dir_comp = ($cwd
+    | str replace $nu.home-path '~'
+    | split row (char separator)
+    )
+
+  if ($dir_comp | length) > 5 {
+    let first = ($dir_comp | first)
+    let last = ($dir_comp | last)
+    let body = (
+      $dir_comp
+      |range 1..-2
+      |each {|x| $x | str substring ..2 }
+      )
+    $dir_comp = ([$first $body $last] | flatten)
+  }
+
+  let style = if $to_home.related == '>' {
+    (ansi xterm_gold3b)
+  } else {
+    ''
+  }
+  $"($style)($dir_comp | str join (char separator))"
+}
+
+# Get repository status as structured data
+export def "git_status" [] {
+  mut status = {
+    in_git_repo                     : false
+    tracking_upstream_branch        : false
+    on_named_branch                 : false
+    commits_ahead                   : 0
+    commits_behind                  : 0
+    upstream_exists_on_remote       : false
+    has_staging_or_worktree_changes : false
+    has_untracked_files             : false
+    has_unresolved_merge_conflicts  : false
+    staging_added_count             : 0
+    staging_modified_count          : 0
+    staging_deleted_count           : 0
+    untracked_count                 : 0
+    worktree_modified_count         : 0
+    worktree_deleted_count          : 0
+    merge_conflict_count            : 0
+  }
```

Comment:
This comment is just to share information. I'm not asking for a change here unless you just want to. Nushell does have a plugin called `gstat` that is meant to capture the git status. Here's a listing of what it reports when you're in a folder with a git repo.
```
gstat
╭─────────────────────┬─────────────╮
│ idx_added_staged    │ 0           │
│ idx_modified_staged │ 0           │
│ idx_deleted_staged  │ 0           │
│ idx_renamed         │ 0           │
│ idx_type_changed    │ 0           │
│ wt_untracked        │ 0           │
│ wt_modified         │ 0           │
│ wt_deleted          │ 0           │
│ wt_type_changed     │ 0           │
│ wt_renamed          │ 0           │
│ ignored             │ 0           │
│ conflicts           │ 0           │
│ ahead               │ 0           │
│ behind              │ 0           │
│ stashes             │ 0           │
│ repo_name           │ nu_scripts  │
│ tag                 │ no_tag      │
│ branch              │ main        │
│ remote              │ origin/main │
╰─────────────────────┴─────────────╯
```

Reply:
cool! i will update this